Server/src/models/DataQueryModel.ts

```diff
@@ -0,0 +1,124 @@
+
+        const connection = getConnection();
+        console.log("Getting data set info based on author");
+        let authorDatasetData: IDatasetModel[] = await selectDatasetIdsQuery(connection.manager)
+            .innerJoin(Publications, 'publication', 'dataset.publicationId = publication.id')
+            .innerJoin('publication.authors', 'author')
+            .where("(author.lastName = :firstNameRef OR author.lastName = :lastNameRef)")
+            .andWhere("(author.firstName = :firstNameRef OR author.firstName = :lastNameRef)")
+            .setParameters({ firstNameRef: firstName, lastNameRef: lastName })
+            .getRawMany();
+        return authorDatasetData;
+    }
+
+    async getDatasetIDFromCategory(category: number): Promise<IDatasetModel[]> {
+
+        const connection = getConnection();
+        console.log("Getting data set info based on category");
+        let categoryDatasetData: IDatasetModel[] = await selectDatasetIdsQuery(connection.manager)
+            .innerJoin(Category, 'category', 'dataset.categoryId = category.id')
+            .where('category.id = :categoryId', { categoryId: category })
+            .getRawMany();
+        return categoryDatasetData;
+    }
+
+    async getDatasetIDFromSubcategory(category: number, subcategory: number): Promise<IDatasetModel[]> {
+
+        const connection = getConnection();
+        console.log("Getting data set info based on subcategory");
+        let subcategoryDatasetData: IDatasetModel[] = await selectDatasetIdsQuery(connection.manager)
+            .innerJoin(Category, 'category', 'dataset.categoryId = category.id')
+            .innerJoin(Subcategory, 'subcategory', 'dataset.subcategoryId = subcategory.id')
+            .where('category.id = :categoryId', { categoryId: category })
+            .andWhere('subcategory.id = :subcategoryId', { subcategoryId: subcategory })
+            .getRawMany();
+        return subcategoryDatasetData;
+    }
+    async getAllData(id: number): Promise<IDatasetResponseModel> {
+
+        const connection = getConnection();
+        console.log("Getting all data sets' raw data");
+
+        console.log("Getting publications");
+        let publicationData: IPublicationModel[] = await selectPublicationsQuery(connection.manager, id)
+
+        console.log("Getting authors");
+        let authorData: IAuthorModel[] = await selectAuthorsQuery(connection.manager, id)
+
+        console.log("Getting data sets");
+        let datasetData: IDatasetModel[] = await selectDatasetsQuery(connection.manager, id)
+
+        console.log("Getting data point data");
+        let datapointData: IDataPointModel[] = await selectDataPointsQuery(connection.manager, id)
+
+        console.log("Getting data set materials");
+        let materialData: IMaterialModel[] = await selectMaterialQuery(connection.manager, id)
+
+        console.log("Getting data point comments");

```

Comment:
Will the console logs remain or will they be removed and replaced by comments? Are they for testing or another reason?

Reply:
They were for testing, I will remove them shortly

Reply:
Done in commit 6066590